client/ayon_houdini/plugins/load/load_filepath.py

```diff
@@ -128,3 +136,111 @@ def format_path(path: str, representation: dict) -> str:
             path = re.sub(pattern, ".{}.{}".format(token, ext), path)
 
         return os.path.normpath(path).replace("\\", "/")
+
+
+class NodePresetLoader(FilePathLoader):
+    """Load node presets.
+
+    It works the same as FilePathLoader, except its extra parameters,
+        2 buttons and target node field.
+        Buttons are used apply the node preset to the target node.
+    """
+
+    label = "Load Node Preset"
+    order = 9
+    icon = "link"
+    color = "white"
+    product_types = {"node_preset"}
+    representations = {"json"}
+
+    # TODO: 
+    #  1. Find a way to cache the node preset, instead of reading the file every time.
+    #  2. Notify the user with the results of Apply button (succeeded, failed and why).
+    #  Note:
+    #    So far we manage the node preset, but we don't manage setting the node preset.
+    def _add_more_node_params(self, attr_folder, node):
+        # allow subclasses to add more params.
+
+        operatore_template = hou.StringParmTemplate(
+            name="target_node",
+            label="Target Node",
+            num_components=1,
+            default_value=("",),
+            string_type=hou.stringParmType.NodeReference,
+            tags= { 
+                "oprelative" : ".",
+                "script_callback" : """
+import json
+from ayon_houdini.api.lib import load_node_preset
+
+json_path = hou.parm("./filepath").eval()
+target_node = hou.parm("./target_node").evalAsNode()
+node_preset = {}
+with open(json_path, "r") as f:
+    node_preset = json.load(f)
+
+node_type = node_preset["metadata"]["type"]
+
+hou.pwd().setColor(hou.Color(0.7, 0.8, 0.87))
+hou.pwd().setComment("")
+hou.pwd().setGenericFlag(hou.nodeFlag.DisplayComment, True)
+if target_node and target_node.type().name() != node_type:
+    hou.pwd().setColor(hou.Color(0.8, 0.45, 0.1))
+    hou.pwd().setComment(
+        f"Target Node type '{target_node.type().name()}' doesn't match the loaded preset type '{node_type}'."
+        "Please note, Applying the preset skips parameters that doesn't exist"
+    )
```

Comment:
Should we make this just a call to a function inside `ayon_houdini.api.lib` so that we can update it from the codebase, instead of storing the code on the node in the actual scenes?

So that we just have `lib.apply_node_preset(filepath)` or `lib.apply_node_preset(json_data)` or alike.

Reply:
Very good point!
Totally yes. let me do it.

Reply:
https://github.com/ynput/ayon-houdini/pull/70/commits/3818b57bda704dce2a41f44e149c6e9bf228dba2